File: grabbot-client-app/src/app/Model/PaginatedBooksResult.ts
export interface PaginatedBooksResult {
    currentPage: number;
    totalPages: number;
    totalCount: number;
    pageSize: number;
    hasPreviousPage: boolean;
    hasNextPage: boolean;
    booksWithPrices: BookWithPriceList[];
}

export interface BookWithPriceList {
    eAN: string;
    title: string;
    sitePrices: WebsitePrice[];
}

export interface WebsitePrice {
    productUrl: string;
    price: number;
    difference: number;
    website: string;
    currency: string;
    lastCrawled: string;
    availability: string;
}

File: BooksBot.API/BooksBot.API/Models/PaginatedBooksResult.cs
using System;
using System.Collections.Generic;

namespace BooksBot.API.Models
{
    public class PaginatedBooksResult
    {  
        public int CurrentPage { get; set; }

        public int TotalPages => TotalCount > PageSize? TotalCount / PageSize : 0 ;

        public int TotalCount { get; set; }
        public int PageSize { get; set; }

        public bool HasPreviousPage => CurrentPage > 1;

        public bool HasNextPage => CurrentPage < TotalPages;

        public List<BookWithPriceList> BooksWithPrices { get; set; } = new();
    }

}
